pkg/profile/artifact.go

```diff
@@ -184,17 +192,21 @@ func (p *Profile) MakeOwnerlessArtifacts() ([]runtime.Object, error) {
 		}
 		switch artifact.Kind {
 		case profilesv1.ProfileKind:
-			if artifact.Profile.URL == p.subscription.Spec.ProfileURL && artifact.Profile.Branch == p.subscription.Spec.Branch {
-				return nil, fmt.Errorf("profile cannot contain profile artifact pointing to itself")
-			}
 			nestedProfileDef, err := getProfileDefinition(artifact.Profile.URL, artifact.Profile.Branch, p.log)
 			if err != nil {
 				return nil, fmt.Errorf("failed to fetch profile %q: %w", artifact.Name, err)
 			}
 			nestedProfile := p.subscription.DeepCopyObject().(*profilesv1.ProfileSubscription)
 			nestedProfile.Spec.ProfileURL = artifact.Profile.URL
 			nestedProfile.Spec.Branch = artifact.Profile.Branch
-			nestedObjs, err := New(p.ctx, nestedProfileDef, *nestedProfile, p.client, p.log).MakeOwnerlessArtifacts()
+
+			nestedSub := New(p.ctx, nestedProfileDef, *nestedProfile, p.client, p.log)
+			profileRepoName := nestedSub.profileRepo()
+			if containsKey(profileRepos, profileRepoName) {
+				return nil, fmt.Errorf("profile cannot contain profile artifact pointing to itself")
```

Comment:
Should we re-word this error message bit? The person seeing this error is likely to not be the person who created the looping profile, so reading `"profile cannot..."` is like "well i didn't make it".

Maybe
```suggestion
				return nil, fmt.Errorf("recursive profile detected: profile cannot contain artifact pointing to itself")
```

so at least it sounds like it is not the fault of the installer? idk

Reply:
Yeah, I suspect we need to output which profile is being recursively included.

In the case of _A -> B -> C -> B_ it might not be obvious what "itself" means?

Reply:
:ok: both changes sound good! I will update :smile: